tensornetwork/tests/network_components_free_test.py

```diff
@@ -862,3 +863,20 @@ def test_save_load_nodes(backend, tmp_path):
   trace = tn.contract_trace_edges(nodes[2])
   loaded_trace = tn.contract_trace_edges(loaded_nodes[2])
   np.testing.assert_allclose(trace.tensor, loaded_trace.tensor)
+
+
+def test_add_to_node_collection_list():
+  container = []
+  with NodeCollection(container):
+    a = Node(np.eye(2))
+    b = Node(np.eye(3))
+
+  assert container == [a, b]
+
+def test_add_to_node_collection_set():
+  container = set()
+  with NodeCollection(container):
+    a = Node(np.eye(2))
+    b = Node(np.eye(3))
+
+  assert container == {a, b}
```

Comment:
Can you add these tests?

1. Test creating a `CopyNode`
2. Test with nested `with` statements

Reply:
added the tests.